crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -589,6 +591,16 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
     let m_db = state.clone().store;
     let m_payment_attempt_update = payment_attempt_update.clone();
     let m_payment_attempt = payment_attempt.clone();
+
+    let payment_attempt = match payment_attempt_update {
+        Some(payment_attempt_update) => PaymentAttempt::from_storage_model(
+            payment_attempt_update
+                .to_storage_model()
+                .apply_changeset(payment_attempt.clone().to_storage_model()),
+        ),
+        None => payment_attempt,
+    };
```

Comment:
```suggestion
    let payment_attempt = payment_attempt_update.map(|payment_attempt_update|{
        PaymentAttempt::from_storage_model(
            payment_attempt_update
                .to_storage_model()
                .apply_changeset(payment_attempt.clone().to_storage_model()),
        )
    });
```

Reply:
map would apply function inside Option and return Option back right, want a default value here as well, in case of None

Reply:
You can use `unwrap_or_else(|f|)`